### PR TITLE
React/week3

### PR DESCRIPTION
**Pull request by CharlesKimMaina**

### CODE REVIEW

File: react/week3/timeregistration/src/EmployeeSearch.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+function EmployeeSearch() {
+    
+    const [search, setSearch] = useState([]);
+    return (
+        <div>
+          <input
+              type="search"
+              placeholder="Search by Name"
+              onChange={(e) => {
+                if (search.length !== 0) {
```
Comment (piyushpalod) [inline react/week3/timeregistration/src/EmployeeSearch.js:16]: Just a suggestion here , better take this logic out in separate function and call the function name here.

File: react/week3/timeregistration/src/RegistrationDetails.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (piyushpalod) [inline react/week3/timeregistration/src/RegistrationDetails.js:3]: what is this prop-types here and whats the use ?

File: react/week3/timeregistration/src/TimeCalculator.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
```
Comment (piyushpalod) [inline react/week3/timeregistration/src/TimeCalculator.js:2]: same here whats the prop-types and whats the use of it ?

File: react/week3/todoform/src/Modal/TodoList.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from 'react'
+import Todos from './Todos';
+import axios from 'axios'
+
+function TodoList() {
+    const [todos, setTodos] = useState([]);
+    
+    useEffect(()=> {
+        axios.get('https://gist.githubusercontent.com/benna100/391eee7a119b50bd2c5960ab51622532/raw')
```
Comment (piyushpalod) [inline react/week3/todoform/src/Modal/TodoList.js:9]: why you are not using async to call API as this may impact performance issue.

**Review by piyushpalod — COMMENTED** [review]
This project contains both NodeJs and React homework, Please separate it out.

File: react/week3/timeregistration/src/RegistrationDetails.js (hunk 2 above)
Comment (CharlesKimMaina) [reply react/week3/timeregistration/src/RegistrationDetails.js:3]: The homework stated that we must demostrate that we know how to use prototypes on components eg proptypes.shape, proptypes.strings, proptypes.number,.,.see "Technical details
All components should have prop types defined."

File: react/week3/timeregistration/src/TimeCalculator.js (hunk 3 above)
Comment (CharlesKimMaina) [reply react/week3/timeregistration/src/TimeCalculator.js:2]: The homework stated that we must demostrate that we know how to use prototypes on components eg proptypes.shape, proptypes.strings, proptypes.number,.,.see "Technical details
All components should have prop types defined."

File: react/week3/todoform/src/Modal/TodoList.js (hunk 4 above)
Comment (CharlesKimMaina) [reply react/week3/todoform/src/Modal/TodoList.js:9]: Great, I have changed Axios n used aync await function,.,.in the reviewed PR,.Thanks

**Pull request merged**